actionpack/test/controller/new_base/render_streaming_test.rb

```diff
@@ -45,44 +45,78 @@ def explicit_cache
 
   class StreamingTest < Rack::TestCase
     test "rendering with streaming enabled at the class level" do
+      env = Rack::MockRequest.env_for("/render_streaming/basic/hello_world")
```

Comment:
Why can't use keep using `get` anymore?

Reply:
Because it doesn't return the raw body or individual chunks and instead wraps it in a `MockResponse` which unfortunately tries to buffer the response.

https://github.com/rack/rack-test/blob/35b73103987828aa228824a6521453a146adff88/lib/rack/test.rb#L366

This line will buffer the response body if possible, which makes it impossible to correctly test streaming. I think we will try to improve testing streaming responses in `rack-test` later on (maybe a flag for "don't buffer the response body").

Reply:
I considered how I could simplify the tests.

Unfortunately, Rails is using it's own bespoke implementation for `TestResponse`:

```ruby
module ActionDispatch
  class TestResponse < Response
    def self.from_response(response)
      new response.status, response.headers, response.body
    end
```

This assigns using `Response#body=`:

```ruby
module ActionDispatch
  class Response
    # Allows you to manually set or override the response body.
    def body=(body)
      if body.respond_to?(:to_path)
        @stream = body
      else
        synchronize do
          @stream = build_buffer self, munge_body_object(body)
        end
      end
    end
```

Unfortunately, at this time, there is no way to bypass `build_buffer self, munge_body_object(body)`. It's necessary to see the unbuffered "chunks" in order to verify streaming. Once this PR, and the other linked PR is merged, I'll try to rework these tests with an improved approach.

I other words, I do agree we should simplify the tests, but I don't want this PR to introduce significant changes to the implementation in order to do so.